internal/translation/deployment/conversion.go

```diff
@@ -901,18 +916,21 @@ func replicationSpecToAtlas(replicationSpecs []*akov2.AdvancedReplicationSpec) *
 
 		specs = append(
 			specs,
-			admin.ReplicationSpec{
-				NumShards:     pointer.NonZeroOrDefault(spec.NumShards, 1),
+			admin.ReplicationSpec20240805{
 				ZoneName:      pointer.MakePtrOrNil(spec.ZoneName),
 				RegionConfigs: &regionConfigs,
 			},
 		)
 	}
 
+	for i := 1; i < replicationSpecs[0].NumShards; i++ {
+		specs = append(specs, specs[0])
+	}
```

Comment:
@roothorp as discussed during the standup, let's verify if this is relevant for geoshareded as well.